lib/codecs/tests/bin/generate-avro-fixtures.rs

```diff
@@ -0,0 +1,490 @@
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    struct Test {
+        long_field: i64,
+    }
+    let value = Test {
+        long_field: 42949672960i64,
+    };
+    generate_test_case(schema, value, "long");
+}
+
+fn generate_avro_test_case_float() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "float_field", "type": "float", "default": 0}
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    struct Test {
+        float_field: f32,
+    }
+    let value = Test {
+        float_field: 123.456,
+    };
+    generate_test_case(schema, value, "float");
+}
+
+fn generate_avro_test_case_double() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "double_field", "type": "double", "default": 0}
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    struct Test {
+        double_field: f64,
+    }
+    let value = Test {
+        double_field: 123.456f64,
+    };
+    generate_test_case(schema, value, "double");
+}
+
+fn generate_avro_test_case_bytes() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "bytes_field", "type": "bytes"}
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    struct Test {
+        bytes_field: Vec<u8>,
+    }
+    let value = Test {
+        bytes_field: vec![1, 2, 3, 4, 5, 6, 6, 7],
+    };
+    generate_test_case(schema, value, "bytes");
+}
+
+fn generate_avro_test_case_string() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "string_field", "type": "string"}
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    struct Test {
+        string_field: String,
+    }
+    let value = Test {
+        string_field: "hello world!".to_string(),
+    };
+    generate_test_case(schema, value, "string");
+}
+
+fn generate_avro_test_case_fixed() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "fixed_field", "type":"fixed", "size": 16}
+        ]
+    }
+    "#;
+    let record = Value::Record(vec![(
+        "fixed_field".into(),
+        Value::Fixed(16, b"1019181716151413".to_vec()),
+    )]);
+    generate_test_case_from_value(schema, record, "fixed");
+}
+
+fn generate_avro_test_case_enum() {
+    let schema = r#"
+    {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "enum_field", "type": "enum", "symbols" : ["SPADES", "HEARTS", "DIAMONDS", "CLUBS"]}
+        ]
+    }
+    "#;
+    #[derive(Debug, Serialize, Deserialize, Clone)]
+    enum Value {
+        SPADES,
+        HEARTS,
+        DIAMONDS,
+        CLUBS,
+    }
```

Comment:
@pront  This is failing `cargo vdev check rust --clippy` with `error: name `SPADES` contains a capitalized acronym`

Should i just change the symbols to this error does not get generated? Or should i look for a way to ignore it?

Reply:
You need to switch to camel case. see https://rust-lang.github.io/api-guidelines/naming.html.

Reply:
Ye but they are generated from this test
```
    let schema = r#"
    {
        "type": "record",
        "name": "test",
        "fields": [
            {"name": "enum_field", "type": "enum", "symbols" : ["SPADES", "HEARTS", "DIAMONDS", "CLUBS"]}
        ]
    }
    "#;
```

`lib/codecs/tests/bin/generate-avro-fixtures.rs` +162

Reply:
You should be able to rename them and run the generation script again.